Upstream: Fixed in master, commit 3c487e7ed54bcfe9c83cb4cbb66549c5d42692fe

From 48db572a33777a6324d8dc6d50513b8118ec06ff Mon Sep 17 00:00:00 2001
From: Benedikt Morbach <benedikt.morbach@googlemail.com>
Date: Sun, 12 Oct 2014 10:11:17 +0200
Subject: [PATCH] tests/scripts/features/archives: pass correct $AR
Organization: Exherbo

621e9ed made sure the test expected the $AR that was detected by
configure. However, it didn't actually pass it to the tested make
instance, which breaks for $AR != ar
---
 tests/scripts/features/archives | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/tests/scripts/features/archives b/tests/scripts/features/archives
index b0acfec..64a3618 100644
--- a/tests/scripts/features/archives
+++ b/tests/scripts/features/archives
@@ -28,17 +28,17 @@ unlink('libxx.a');
 
 # Very simple
 run_make_test('all: libxx.a(a1.o)',
-              '', "$ar rv libxx.a a1.o\n$created");
+              "AR=$ar", "$ar rv libxx.a a1.o\n$created");
 
 # Multiple .o's.  Add a new one to the existing library
 ($_ = $add) =~ s/#OBJECT#/a2.o/g;
 run_make_test('all: libxx.a(a1.o a2.o)',
-              '', "$ar rv libxx.a a2.o\n$_");
+              "AR=$ar", "$ar rv libxx.a a2.o\n$_");
 
 # Touch one of the .o's so it's rebuilt
 utouch(-40, 'a1.o');
 ($_ = $repl) =~ s/#OBJECT#/a1.o/g;
-run_make_test(undef, '', "$ar rv libxx.a a1.o\n$_");
+run_make_test(undef, "AR=$ar", "$ar rv libxx.a a1.o\n$_");
 
 # Use wildcards
 run_make_test('all: libxx.a(*.o)',
@@ -47,20 +47,20 @@ run_make_test('all: libxx.a(*.o)',
 # Touch one of the .o's so it's rebuilt
 utouch(-30, 'a1.o');
 ($_ = $repl) =~ s/#OBJECT#/a1.o/g;
-run_make_test(undef, '', "$ar rv libxx.a a1.o\n$_");
+run_make_test(undef, "AR=$ar", "$ar rv libxx.a a1.o\n$_");
 
 # Use both wildcards and simple names
 utouch(-50, 'a2.o');
 ($_ = $add) =~ s/#OBJECT#/a3.o/g;
 $_ .= "$ar rv libxx.a a2.o\n";
 ($_ .= $repl) =~ s/#OBJECT#/a2.o/g;
-run_make_test('all: libxx.a(a3.o *.o)', '',
+run_make_test('all: libxx.a(a3.o *.o)', "AR=$ar",
               "$ar rv libxx.a a3.o\n$_");
 
 # Check whitespace handling
 utouch(-40, 'a2.o');
 ($_ = $repl) =~ s/#OBJECT#/a2.o/g;
-run_make_test('all: libxx.a(  a3.o    *.o     )', '',
+run_make_test('all: libxx.a(  a3.o    *.o     )', "AR=$ar",
               "$ar rv libxx.a a2.o\n$_");
 
 rmfiles(qw(a1.o a2.o a3.o libxx.a));
@@ -86,7 +86,7 @@ default: lib(foo)
 (%): %.vhd ; @cd $(DIR) && touch $(*F) && $(AR) $(ARFLAGS) $@ $(*F) >/dev/null 2>&1 && rm $(*F)
 .PHONY: default
 !,
-              '', "");
+              "AR=$ar", "");
 
 run_make_test(undef, '', "#MAKE#: Nothing to be done for 'default'.\n");
 
-- 
2.1.2

